libs/pyrt_wit_wire/main.py

```diff
@@ -60,8 +77,103 @@ def op_to_handler(op: MatInfo) -> ErasedHandler:
         exec(data_parsed["source"], module.__dict__)
         fn = module.__dict__[data_parsed["func_name"]]
         return ErasedHandler(handler_fn=lambda inp: fn(inp))
+    elif data_parsed["ty"] == "import_function":
+        prefix = data_parsed["func_name"]
+
+        modules_raw = data_parsed["sources"]
+        finder = ThePathFinder(
+            {os.path.join(prefix, path): modules_raw[path] for path in modules_raw}
+        )
+        sys.meta_path.append(finder)
+
+        module = importlib.import_module(
+            ThePathFinder.path_to_module(
+                os.path.join(prefix, data_parsed["root_src_path"])
+            )
+        )
+        return ErasedHandler(handler_fn=getattr(module, data_parsed["func_name"]))
     elif data_parsed["ty"] == "lambda":
         fn = eval(data_parsed["source"])
         return ErasedHandler(handler_fn=lambda inp: fn(inp))
     else:
         raise Err(InitError_UnexpectedMat(op))
+
+
+class ThePathFinder(importlib.abc.MetaPathFinder):
+    @staticmethod
+    def path_to_module(path: str):
+        return os.path.splitext((os.path.normpath(path)))[0].replace("/", ".")
+
+    def __init__(self, modules: Dict[str, str]):
+        self._mods_raw = modules
+        self._pkgs = set()
+        for path in self._mods_raw:
+            dirname = os.path.dirname(path)
+            while dirname != "/" and dirname != "" and dirname != ".":
+                if dirname not in self._mods_raw:
+                    self._pkgs.add(dirname)
+                dirname = os.path.dirname(dirname)
+        self._mod_names = {ThePathFinder.path_to_module(path): path for path in modules}
+        self._pkg_names = {
+            ThePathFinder.path_to_module(path): path for path in self._pkgs
+        }
+
+    # Look for a spec under a certain module name
+    # https://peps.python.org/pep-0302/
+    # https://peps.python.org/pep-0451/
```

Comment:
What I'm seeing in these PEPs and the docs imply the limitation/breaking change I've outlined in the PR desc is not necessary but I was not able to get it working despite hours.

Some context links:
- https://stackoverflow.com/questions/41941079/what-is-the-difference-between-sys-meta-path-and-sys-path-hooks-importer-obj
- https://stackoverflow.com/questions/41990169/how-to-use-sys-path-hooks-for-customized-loading-of-modules